### ISSUE TITLE
Virtual property causes incorrect MySQL query

### ISSUE THREAD

**Issue by Asesjix** [mikro-orm/mikro-orm#5261]
### Describe the bug

When specifying a virtual property in the field options of the find function, the virtual property is erroneously included in the MySQL database query. This is problematic because virtual properties do not exist in the database, leading to incorrect query results or errors. Additionally, when the virtual property is omitted from the field options, the return type of the find function is no longer accurate, potentially causing issues with type checking and data handling downstream.

```mysql
select `p0`.`id`, `p0`.`rateSum`, `p0`.`rateCount`, `p0`.`rating` from `my_entity` as `p0` where `p0`.`id` = 1 limit 1 - Unknown column 'rating' in 'field list'
```

### Reproduction

Define a virtual property in the model.
Use the find function with the virtual property specified in the field options.
Observe that the virtual property is included in the MySQL query.

```ts
export class MyEntity {
  id!: number
  rateSum?: number = 0
  rateCount?: number = 0
  likeCount = 0
  mtime!: Date

  get rating() {
    if (
      !this.rateCount ||
      this.rateCount <= MIN_VOTES_FOR_RATING ||
      !this.rateSum ||
      Math.abs(this.rateSum) > this.rateCount
    ) {
      return 0
    }
    const average = (this.rateSum / this.rateCount) * 100
    return Math.round(average)
  }
}

export const MyEntitySchema = new EntitySchema({
  class: MyEntity,
  properties: {
    id: { primary: true, type: 'number' },
    rateSum: {
      type: 'number',
      columnType: 'mediumint',
      nullable: true,
      default: 0,
    },
    rateCount: {
      type: 'number',
      columnType: 'mediumint',
      nullable: true,
      default: 0,
    },
    rating: {
      type: 'method',
      persist: false,
      getter: true,
    },
    mtime: {
      type: 'Date',
      defaultRaw: `CURRENT_TIMESTAMP`,
    },
    likeCount: { type: 'number', default: 0 },
  },
})

async getMyEntity(id: number): Promise<MyEntityDto | null> {
  const myEntity = await this.#entityRepo.findOne(
    { id },
    {
      fields: ['id', 'rateSum', 'rateCount', 'rating']
    },
  )

  return myEntity ? mapToMyEntityDto(myEntity) : null
}
```

### What driver are you using?

@mikro-orm/mysql

### MikroORM version

6.1.4

### Node.js version

18.18.2

### Operating system

Debian

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Why do you put it into the `fields` hint? I don't think we prune that array, and it feels like a breaking change to start doing so. Sounds like a wontfix.

**Comment by Asesjix** [comment]
I added this info in the description:

When the virtual property is omitted from the field options, the return type of the find function is no longer accurate, potentially causing issues with type checking and data handling downstream.

**Comment by B4nan** [comment]
I see, and I'm afraid this might also mean it will not be part of the serialized form. 

And this worked in v5 or you had to add it when trying to migrate to v6?

**Comment by StevenKowalzik** [comment]
Yes, in v5 the returned type contained the virtual property _rating_ without it being defined in the fields array.

**Comment by Asesjix** [comment]
I have checked this again. Previously, the rating was not specified in the fields and was therefore not included in the database query. However, the return type of find was not a problem because the property was there, just like all the others.

Now, with version 6, the property is no longer in the return type of find. Therefore, we tried to include the property in the fields, but this causes the error with the query.
